### PR TITLE
Fix PTM stable latest related to the default.qubit.legacy

### PR DESCRIPTION
**Pull request by LuisAlfredoNu**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [X] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
The CIs on the PTM fail due to incompatibilities between deprecated devices.

**Description of the Change:**

The PTM has an unexpected behaviour for Stable Lightnin - Latest PennyLane because on CIs still using the device default.qubit.legacy. The solution is patching tests following deprecation which will be removed after v0.39.

**Benefits:**
No more errors in the PTM stable-latest regarding the `default.qubit.legacy`

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-76902]

### CODE REVIEW

**Comment by LuisAlfredoNu** [comment]
> Awesome work @LuisAlfredoNu ! Would you please add an entry to the changelog? Thanks

Done! :smile:

**Review by multiphaseCFD — APPROVED** [review]
Thanks @LuisAlfredoNu ! LGTM

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -46,6 +46,9 @@
 
 ### Improvements
 
+*  Fix PTM stable latest related to the default.qubit.legacy.
```
Comment (maliasadi) [inline .github/CHANGELOG.md:49]:
```suggestion
*  Fix PTM stable latest related to `default.qubit.legacy`. 
```

Comment (maliasadi) [reply .github/CHANGELOG.md:49]: Shouldn't this be listed under "Bug fixes"?

**Review by maliasadi — APPROVED** [review]
Thanks @LuisAlfredoNu!

File: .github/CHANGELOG.md (hunk 1 above)
Comment (LuisAlfredoNu) [reply .github/CHANGELOG.md:49]: You are right! I will move it and disable all labels for a quick merge after rerunning the minimum test batch.

**Pull request merged**